hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1725,6 +1739,189 @@ public void delete(final Path path, final boolean recursive,
+   * only blob in the directory, it will render path as non-existing. To prevent
+   * that happening, client create marker-based directory.
+   *
+   * @param path path getting deleted
+   * @param tracingContext tracingContext to trace the API flow
+   * @throws IOException
+   */
+  private void createParentDirectory(final Path path,
+      final TracingContext tracingContext) throws IOException {
+    if (path.isRoot()) {
+      return;
+    }
+    Path parentPath = path.getParent();
+    if (parentPath.isRoot()) {
+      return;
+    }
+
+    String srcPathStr = path.toUri().getPath();
+    String srcParentPathSrc = parentPath.toUri().getPath();
+    LOG.debug(String.format(
+        "Creating Parent of Path %s : %s", srcPathStr, srcParentPathSrc));
+    createDirectoryMarkerBlob(parentPath, null, FsPermission.getDirDefault(),
+        FsPermission.getUMask(
+            getAbfsConfiguration().getRawConfiguration()),
+        tracingContext, true);
+    LOG.debug(String.format("Directory for parent of Path %s : %s created",
+        srcPathStr, srcParentPathSrc));
+  }
+
+  /**
+   * Consumes list of blob over the consumer. Deletes the blob listed.
+   * The deletion of the consumed blobs is executed over parallel threads spawned
+   * by an ExecutorService with threads equal to {@link AbfsConfiguration#getBlobDirDeleteMaxThread()}.
+   * Once a list of blobs that were consumed are deleted, the next batch of blobs
+   * are consumed.
+   *
+   * @param srcPath path of the directory which has to be deleted
+   * @param consumer {@link ListBlobConsumer} object through which batches of blob
+   * are consumed.
+   * @param tracingContext object for tracing the API flow
+   *
+   * @throws AzureBlobFileSystemException exception received from server call.
+   */
+  private void deleteOnConsumedBlobs(final Path srcPath,
+      final ListBlobConsumer consumer,
+      final TracingContext tracingContext) throws AzureBlobFileSystemException {
+    ExecutorService deleteBlobExecutorService = Executors.newFixedThreadPool(
+        getAbfsConfiguration().getBlobDirDeleteMaxThread());
+    try {
+      while (!consumer.isCompleted()) {
+        final List<BlobProperty> blobList = consumer.consume();
+        if (blobList == null) {
+          continue;
+        }
+        List<Future> futureList = new ArrayList<>();
+        for (BlobProperty blobProperty : blobList) {
+          futureList.add(deleteBlobExecutorService.submit(() -> {
+            String blobPropertyPathStr = blobProperty.getPath()
+                .toUri()
+                .getPath();
+            try {
+              LOG.debug(String.format("Deleting Path %s", blobPropertyPathStr));
+              client.deleteBlobPath(blobProperty.getPath(), null,
+                  tracingContext);
+              LOG.debug(String.format("Deleted Path %s", blobPropertyPathStr));
+            } catch (AzureBlobFileSystemException ex) {
+              if (ex instanceof AbfsRestOperationException
```

Comment:
Excessive logging here too

Reply:
Agree, too much of logs would happen, since its for all child blobs. Taken.